src/apps/relay/mainrelay.c

```diff
@@ -2779,1082 +2968,1086 @@ static void print_features(unsigned long mfn) {
-      TURN_LOG_FUNC(TURN_LOG_LEVEL_ERROR, "Cannot start daemon process\n");
-      exit(-1);
+    if (!dh) {
+      if (turn_params.dh_key_size == DH_566)
+        dh = get_dh566();
+      else if (turn_params.dh_key_size == DH_1066)
+        dh = get_dh1066();
+      else
+        dh = get_dh2066();
     }
-#else
-    if (daemon(1, 0) < 0) {
-      TURN_LOG_FUNC(TURN_LOG_LEVEL_ERROR, "Cannot start daemon process\n");
-      exit(-1);
+
+    if (!dh) {
+      TURN_LOG_FUNC(TURN_LOG_LEVEL_ERROR, "%s: ERROR: cannot allocate DH suite\n", __FUNCTION__);
+      err = 1;
+    } else {
+      if (1 != SSL_CTX_set_tmp_dh(ctx, dh)) {
+        TURN_LOG_FUNC(TURN_LOG_LEVEL_ERROR, "%s: ERROR: cannot set DH\n", __FUNCTION__);
+        err = 1;
+      }
+      DH_free(dh);
     }
-    reset_rtpprintf();
-#endif
   }
 
-  if (turn_params.pidfile[0]) {
+  { // secret key
 
-    char s[2049];
-    FILE *f = fopen(turn_params.pidfile, "w");
-    if (f) {
-      STRCPY(s, turn_params.pidfile);
-    } else {
-      TURN_LOG_FUNC(TURN_LOG_LEVEL_WARNING, "Cannot create pid file: %s\n", turn_params.pidfile);
+    if (turn_params.secret_key_file[0]) {
+      FILE *f = fopen(turn_params.secret_key_file, "r");
 
-      {
-        const char *pfs[] = {"/var/run/turnserver.pid",
-                             "/var/spool/turnserver.pid",
-                             "/var/turnserver.pid",
-                             "/var/tmp/turnserver.pid",
-                             "/tmp/turnserver.pid",
-                             "turnserver.pid",
-                             NULL};
-        const char **ppfs = pfs;
-        while (*ppfs) {
-          f = fopen(*ppfs, "w");
-          if (f) {
-            STRCPY(s, *ppfs);
-            break;
-          } else {
-            ++ppfs;
+      if (!f) {
+        perror("Cannot open Secret-Key file");
+      } else {
+        fseek(f, 0, SEEK_SET);
+        rc = fread(turn_params.secret_key, sizeof(char), 16, f);
+        if (rc == 0) {
+          TURN_LOG_FUNC(TURN_LOG_LEVEL_ERROR, "%s: ERROR: Secret-Key file is empty\n", __FUNCTION__);
+        } else {
+          if (rc != 16) {
+            TURN_LOG_FUNC(TURN_LOG_LEVEL_ERROR, "%s: ERROR: Secret-Key length is not enough\n", __FUNCTION__);
           }
         }
+        fclose(f);
       }
     }
-
-    if (f) {
-      fprintf(f, "%lu\n", (unsigned long)getpid());
-      fclose(f);
-      TURN_LOG_FUNC(TURN_LOG_LEVEL_INFO, "pid file created: %s\n", s);
-    }
   }
-#endif
 
-  setup_server();
-
-#if defined(WINDOWS)
-  // TODO: implement it!!! add windows server
-#else
-  struct event *ev = evsignal_new(turn_params.listener.event_base, SIGUSR2, reload_ssl_certs, NULL);
-  event_add(ev, NULL);
+  {
+    int op = 0;
 
-  ev = evsignal_new(turn_params.listener.event_base, SIGTERM, shutdown_handler, NULL);
-  event_add(ev, NULL);
-  ev = evsignal_new(turn_params.listener.event_base, SIGINT, shutdown_handler, NULL);
-  event_add(ev, NULL);
+#if defined(SSL_OP_NO_SSLv2)
```

Comment:
SSLv2 and v3 are even older than TLS 1.0 and 1.1, which are both deprecated.
Is it a good idea to support old, known insecure, algorithms?